src/ray/pubsub/publisher.cc

```diff
@@ -207,24 +243,30 @@ void Publisher::Publish(const ObjectID &object_id) {
     auto it = subscribers_.find(subscriber_id);
     RAY_CHECK(it != subscribers_.end());
     auto &subscriber = it->second;
-    subscriber->QueueMessage(object_id);
+    subscriber->QueueMessage(pub_message);
   }
 }
 
-bool Publisher::UnregisterSubscriber(const SubscriberID &subscriber_id) {
+bool Publisher::UnregisterSubscription(const rpc::ChannelType channel_type,
+                                       const SubscriberID &subscriber_id,
+                                       const std::string &key_id_binary) {
   absl::MutexLock lock(&mutex_);
-  return UnregisterSubscriberInternal(subscriber_id);
+  return subscription_index_map_[channel_type].EraseEntry(key_id_binary, subscriber_id);
```

Comment:
Is it safe that `subscription_index_map_` has no key `channel_type` and then this line is invoked？
It's inserted an empty entry and then invoke the `empty_entry.EraseEntry()`.

Reply:
Hmm my assumption is people won't do this sort of mistake (or it will be revealed right away if you do so), but I can just add a CHECK instead here.